test/it/controllers/EmailByEoriControllerSpec.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.WireMockIntegrationSpec
+import config.AppConfig
+import connectors.CustomsDataStoreConnector
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Helpers}
+import stubs.MockGenericDownstreamService
+import testdata.TestData
+
+class EmailByEoriControllerSpec extends WireMockIntegrationSpec with MockGenericDownstreamService {
```

Comment:
Tests present in this suite are testing not only the Controller but the whole feature. May I suggest renaming this suite to reflect that? And to write proper unit tests for Controller separately?

Reply:
Sorry, I do not understand :relaxed:  Isn't this an integration test? Would you please suggest a better name?
Ok to write a unit test for the controller only.

Reply:
I meant that these test are not only testing the controller, but rather contoller, connector, appConfig and potentially any other class involved. Therefore, I think there sshould be a separate group of integration tests that would be not class-specific, but maybe feature/flow-specific.